GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/model/GpsData.java

```diff
@@ -15,26 +16,40 @@
     private DecimalFormat locationDF= new DecimalFormat("0.000000");
     private DecimalFormat speedDF = new DecimalFormat("0.0000");
 
+    /**
+     * Constructor to create a GpsData object
+     * @param location
+     */
     public GpsData(Location location) {
         latitude = String.valueOf(locationDF.format(location.getLatitude()));
         longitude = String.valueOf(locationDF.format(location.getLongitude()));
         speed = String.valueOf(location.hasSpeed() ? speedDF.format(location.getSpeed()) : "0.0000");
     }
 
     /**
-     * Get the string of GpsData
-     * @return a string of GpsData
+     * Get the latitude of the GpsData
+     * @return latitude
+     */
+    @NonNull
+    public String getLatitude(){
+        return this.latitude;
+    }
+
+    /**
+     * Get the longitude of the GpsData
+     * @return longitude
+     */
+    @NonNull
+    public String getLongitude(){
+        return this.longitude;
+    }
+
+    /**
+     * Get the speed of the GpsData
+     * @return speed
      */
     @NonNull
-    public String getGpsDataString() {
```

Comment:
lets keep this method for future debugging

Reply:
How about change this method to ` @Override toString() ` method and make GpsData to immutable value object too?

Reply:
Sounds good.

Reply:
Done.